feat(standalone): support JSON format (#12333)
apache/apisix: docs/en/latest/deployment-modes.md

    config_provider: yaml
```

> You can also provide the configuration in JSON format by placing it in `conf/apisix.json`. At this point, you need to change the `deployment.role_data_plane.config_provider` to `json`.

Reviewer: ```suggestion
You can also provide the configuration in JSON format by placing it in `conf/apisix.json`. Before proceeding, you should change the `deployment.role_data_plane.config_provider` to `json`.
```

#END
```

> *WARNING*: APISIX will not load the rules into memory from file `conf/apisix.yaml` if there is no `#END` at the end. However, when using `conf/apisix.json`, the `#END` marker is not required, as APISIX can directly parse and validate the JSON structure.

Reviewer 2: You need to clarify the priority of the `apisix.yaml` and `apisix.json` configuration files.
When both files exist, which one is valid.

#END
```

```json5

> // JSON configuration can be directly translated from YAML, this is just an example.

Author: OK, I will wait for others to finish the feedback on the code part and repair the document part.

If I remember correctly, this might involve converting the document to MDX.
